### PR TITLE
update angular, move most deps to dev, fix lint, remove unused deps

### PR DESCRIPTION
**Pull request by SimonSchick**
Update ng2 to ^2.2.3
fix rxjs to beta12 as that's used by us rn
cleaned npmignore
fixed npm license and repo
rm'd some unused deps
moved some deps to dep deps
fixed lint

build and test pass locally

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,53 +1,2 @@
...
-protractor.conf.js
-tsconfig.json
-tslint.json
-typedoc.json
-webpack.config.js
-.travis.yml
-.jshintrc
-.editorconfig
+/*
+!/dist
```
Comment (connor4312) [inline .npmignore]: We're required to distribute the license too

File: tslint.json [hunk 2]
```diff
@@ -15,6 +15,14 @@
     "no-relative-imports": false,
     "no-stateless-class": false,
     "trailing-comma": [true, {"multiline": "always", "singleline": "never"}],
+    "no-for-in-array": false,
+    "restrict-plus-operands": false,
```
Comment (connor4312) [inline tslint.json]: These can both be enabled here

Comment (SimonSchick) [inline tslint.json]: Not if I want to use it in vscode, we had a discussion about that before and the result is that both these rules don't add much value in the end and it's better to have vscode be able to show these.

Comment (connor4312) [inline tslint.json]: Ok

File: .npmignore (hunk 1 above)
Comment (SimonSchick) [inline .npmignore]: To clarify, npm will ALWAYS include `readme`(and variations) `license` and variations `package.json`

Comment (SimonSchick) [inline .npmignore]: https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package

Comment (connor4312) [inline .npmignore]: Ah, great, I wasn't aware of that

**Pull request merged**